src/lib.rs

```diff
@@ -444,6 +439,30 @@ impl CTRConfig {
             process::exit(status.code().unwrap_or(1));
         }
     }
+
+    /// Possible cases:
+    /// - icon path specified (exit with error if doesn't exist)
+    /// - icon path unspecified, icon.png exists
+    /// - icon path unspecified, icon.png does not exist
+    fn icon_path(&self) -> Utf8PathBuf {
```

Comment:
I don't find this program logic to be very readable and straightforward.

Wouldn't it be better to have a `match` on the icon path given by the user and *then* check for the error cases? Something like:
```rust
match self.icon_path.as_deref() {
     Some(path) => /* check whether the given path is valid and panic if not */,
     None => {
         // Checking for possible default paths:
         let path = Utf8Path::new("icon.png");
         /* check whether the icon.png file exists or use the default icon if not */
     }
}
```

Also, although it's almost beyond the scope for `cargo-3ds`, should we also check whether the `default_icon.png` file exists within the dkP directories? I'm pretty sure `smdhtool` wouldn't like that scenario.

Reply:
I can try to test what the behavior is if the icon is missing... my guess is that it would just fail when we run `smdhtool` instead of here, but it's not hard for us to do an extra check as well.

As for why it's structured like this, I was hoping to avoid duplicating the existence check if I could help it, and I iterated on a couple different things before this.  I'll see if I can improve it to be more like your recommendation

Reply:
Tried to make it a little more readable with a `Result`, let me know what you think

Reply:
Much better :+1:.